src/fsrs_optimizer/fsrs_simulator.py

```diff
@@ -141,13 +143,87 @@ def simulate(
         [1, 2, 3, 4], deck_size, p=first_rating_prob
     )
     card_table[col["rating"]] = card_table[col["rating"]].astype(int)
+    card_table[col["ease"]] = 0
 
     revlogs = {}
     review_cnt_per_day = np.zeros(learn_span)
     learn_cnt_per_day = np.zeros(learn_span)
     memorized_cnt_per_day = np.zeros(learn_span)
     cost_per_day = np.zeros(learn_span)
 
+    # Anki scheduler constants
+    GRADUATING_IVL = 1
+    EASY_IVL = 4
+    NEW_IVL = 0
+    HARD_IVL = 1.2
+    INTERVAL_MODIFIER = 1.0
+    EASY_BONUS = 1.3
+    MIN_IVL = 1
+
+    def anki_sm2_scheduler(interval, real_interval, ease, rating):
+        # Handle new cards (ease == 0)
+        is_new_card = ease == 0
+        new_card_intervals = np.where(rating == 4, EASY_IVL, GRADUATING_IVL)
+        new_card_eases = np.full_like(ease, 2.5)
+
+        # Handle review cards
+        delay = real_interval - interval
+
+        # Calculate intervals for each rating
+        again_interval = np.minimum(
+            np.maximum(
+                np.round(interval * NEW_IVL * INTERVAL_MODIFIER + 0.01),
+                MIN_IVL,
+            ),
+            max_ivl,
+        )
+        hard_interval = np.minimum(
+            np.maximum(
+                np.round(interval * HARD_IVL * INTERVAL_MODIFIER + 0.01),
+                np.maximum(interval + 1, MIN_IVL),
+            ),
+            max_ivl,
+        )
+        good_interval = np.minimum(
+            np.maximum(
+                np.round((interval + delay / 2) * ease * INTERVAL_MODIFIER + 0.01),
+                np.maximum(hard_interval + 1, MIN_IVL),
+            ),
+            max_ivl,
+        )
+        easy_interval = np.minimum(
+            np.maximum(
+                np.round(real_interval * ease * INTERVAL_MODIFIER * EASY_BONUS + 0.01),
+                np.maximum(good_interval + 1, MIN_IVL),
+            ),
+            max_ivl,
+        )
+
+        # Select intervals based on rating
+        review_intervals = np.choose(
+            rating - 1, [again_interval, hard_interval, good_interval, easy_interval]
+        )
+
+        # Calculate new eases
+        review_eases = (
+            np.choose(
+                rating - 1,
+                [
+                    ease - 0.2,
+                    ease - 0.15,
+                    ease,
+                    ease + 0.15,
+                ],
+            ),
+        )
+        review_eases = np.maximum(review_eases, 1.3)
```

Comment:
```suggestion
        review_eases = np.maximum(review_eases, 1.3)[0]
```
No clue why this is needed??????